pkg/ccl/backupccl/restore_planning.go

```diff
@@ -1067,6 +1068,12 @@ func restorePlanHook(
 		return nil, nil, nil, false,
 			errors.New("cannot run RESTORE with schema_only until cluster has fully upgraded to 22.2")
 	}
+
+	if restoreStmt.Options.RemoveRegions && !p.ExecCfg().Settings.Version.IsActive(ctx, clusterversion.V23_1) {
```

Comment:
I don't think this is doing what we think it is doing here: this check can return true -- that is, V23_1 is active -- even if there are nodes in the cluster that do not contain this change. That is, there is no way, in a backport, to be sure before you write or send something that the node that reads/receives it will be running your new code -- it could still be running v23.1.9 or something, but IsActive(V23_1) is true.

Reply:
in a team discussion, we were thinking of introducing a new hidden cluster setting in the backport, which is named something like `restore.remove_regions.enabled`. the new restore option would only be allowed if this setting is on, and if a customer sets it, we'll know they only did so if we advised them to do so and it can serve as implicit confirmation that all nodes are on the new 23.1 patch release.